test/e2e/apiextensions/compositon_test.go

```diff
@@ -0,0 +1,745 @@
+// +build e2e
+
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiextensions
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/utils/pointer"
+
+	corev1 "k8s.io/api/core/v1"
+	kextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+	extv1 "github.com/crossplane/crossplane/apis/apiextensions/v1"
+	v1 "github.com/crossplane/crossplane/apis/pkg/v1"
+)
+
+func TestCompositeResourceGetsReady(t *testing.T) {
+	cases := map[string]struct {
```

Comment:
What do you think about having `args` and `want` structs in this function that has the inputs and outputs of the test so that you write the function calls only once in `t.Run` and use those input/output pairs? An example is [here](https://github.com/crossplane/crossplane-runtime/blob/423918f/pkg/reconciler/managed/api_test.go#L119).

Reply:
I was thinking about it but in these tests, our input would always be a single parameter: ctx, and output requirement will be no errors, so will it be okay to skip it for now 🤔?